megatron/core/models/mamba/mamba_model.py

```diff
@@ -237,14 +352,54 @@ def forward(
             rotary_pos_emb=rotary_pos_emb,
         )
 
+        # Get output weight for shared embeddings
+        output_weight = None
+        if self.share_embeddings_and_output_weights:
+            output_weight = self.shared_embedding_or_output_weight()            
+
+        if self.mtp_process:
+            extra_block_kwargs = None
+            packed_seq_params = None
+            decoder_input, rotary_pos_emb, rotary_pos_cos, rotary_pos_sin, sequence_len_offset = self._preprocess(
+                input_ids=input_ids,
+                position_ids=position_ids,
+                decoder_input=decoder_input,
+                inference_context=inference_context,
+                packed_seq_params=packed_seq_params,
+            )
+
+
+            hidden_states = self.mtp(
+                input_ids=input_ids,
+                position_ids=position_ids,
+                hidden_states=hidden_states,
+                attention_mask=attention_mask,
+                inference_params=inference_params,
+                rotary_pos_emb=rotary_pos_emb,
+                rotary_pos_cos=rotary_pos_cos,
+                rotary_pos_sin=rotary_pos_sin,
+                packed_seq_params=packed_seq_params,
+                sequence_len_offset=sequence_len_offset,
+                embedding=self.embedding,
+                **(extra_block_kwargs or {}),
+            )
+
         if not self.post_process:
```

Comment:
Is it possible for `self.post_process` to be `False` but `self.mtp_process` to be `True`? Not sure that this logic here aligns with the logic in the initializer.